lnwallet/chanfunding/coin_select_test.go

```diff
@@ -42,6 +50,107 @@ func fundingFee(feeRate chainfee.SatPerKWeight, numInput int, // nolint:unparam
 	return feeRate.FeeForWeight(totalWeight)
 }
 
+// TestCalculateFees tests that the helper function to calculate the fees
+// both with and without applying a change output is done correctly for
+// (N)P2WKH inputs, and should raise an error otherwise.
+func TestCalculateFees(t *testing.T) {
+	t.Parallel()
+
+	const feeRate = chainfee.SatPerKWeight(1000)
+
+	type testCase struct {
+		name  string
+		utxos []Coin
+
+		expectedFeeNoChange   btcutil.Amount
+		expectedFeeWithChange btcutil.Amount
+		expectErr             bool
+	}
+
+	testCases := []testCase{
+		{
+			name: "one P2WKH input",
+			utxos: []Coin{
+				{
+					TxOut: wire.TxOut{
+						PkScript: p2wkhScript,
+						Value:    1,
+					},
+				},
+			},
+
+			expectedFeeNoChange:   487,
+			expectedFeeWithChange: 611,
+			expectErr:             false,
+		},
+
+		{
+			name: "one NP2WKH input",
+			utxos: []Coin{
+				{
+					TxOut: wire.TxOut{
+						PkScript: np2wkhScript,
+						Value:    1,
+					},
+				},
+			},
+
+			expectedFeeNoChange:   579,
+			expectedFeeWithChange: 703,
+			expectErr:             false,
+		},
+
+		{
+			name: "not supported P2KH input",
+			utxos: []Coin{
+				{
+					TxOut: wire.TxOut{
+						PkScript: p2khScript,
+						Value:    1,
+					},
+				},
+			},
+
+			expectErr: true,
+		},
+	}
+
+	for _, test := range testCases {
+		test := test
+		t.Run(test.name, func(t *testing.T) {
+			feeNoChange, feeWithChange, err := calculateFees(
+				test.utxos, feeRate,
+			)
+			if !test.expectErr && err != nil {
+				t.Fatalf(err.Error())
+			}
+
+			if test.expectErr && err == nil {
+				t.Fatalf("expected error")
+			}
+
+			// If we got an expected error, there is nothing more
+			// to test.
+			if test.expectErr {
+				return
+			}
+
+			// Check the calculated fee without change output.
+			if feeNoChange != test.expectedFeeNoChange {
+				t.Fatalf("expected feeNoChange %d, got %d",
+					test.expectedFeeNoChange, feeNoChange)
+			}
+
+			// Check the calculated fee with change output.
+			if feeWithChange != test.expectedFeeWithChange {
+				t.Fatalf("expected feeWithChange %d, got %d",
+					test.expectedFeeWithChange,
+					feeWithChange)
```

Comment:
could shorten this all to just `require.Equal(t, test.expectedFee..., fee...)`, the error cases have 0 values anyway so they'd also pass.